srtcore/srt.h

```diff
@@ -300,9 +300,20 @@ typedef enum SRT_TRANSTYPE
 // This is for MPEG TS and it's a default SRTO_PAYLOADSIZE for SRTT_LIVE.
 static const int SRT_LIVE_DEF_PLSIZE = 1316; // = 188*7, recommended for MPEG TS
 
-// This is the maximum payload size for Live mode, should you have a different
-// payload type than MPEG TS.
-static const int SRT_LIVE_MAX_PLSIZE = 1456; // MTU(1500) - UDP.hdr(28) - SRT.hdr(16)
+// DEPRECATED. Use one of these below instead.
+SRT_ATR_DEPRECATED_PX static const int SRT_LIVE_MAX_PLSIZE SRT_ATR_DEPRECATED = 1456; // MTU(1500) - UDP.hdr(28) - SRT.hdr(16)
+
+// These constants define the maximum size of the payload
+// in a single UDP packet, depending on the IP version, and
+// with the default socket options, that is:
+// * default 1500 bytes of MTU (see SRTO_MSS)
+// * without FEC packet filter (see SRTO_PACKETFILTER)
+// * without AEAD through AES-GCM (see SRTO_CRYPTOMODE)
+static const int SRT_MAX_PLSIZE_AF_INET = 1456; // MTU(1500) - IPv4.hdr(20) - UDP.hdr(8) - SRT.hdr(16)
+static const int SRT_MAX_PLSIZE_AF_INET6 = 1444; // MTU(1500) - IPv6.hdr(32) - UDP.hdr(8) - SRT.hdr(16)
+
+// A macro for these above in case when the IP family is passed as a runtime value.
+#define SRT_MAX_PLSIZE(famspec) ((famspec) == AF_INET ? SRT_MAX_PLSIZE_AF_INET : SRT_MAX_PLSIZE_AF_INET6)
```

Comment:
I am hesitant in having these in the API, because the maximum payload size also depends on the MTU size.

Reply:
I was simply trying to do some replacement for the existing the "MAX PLSIZE" constant. Not sure if anyone is using it or whether it is anyhow useful. Note that the live payload size (1316) is still lower than these two, also including FEC and AEAD. So the only use case for these constants is for a case when you have a live stream, but you want to have a bigger payload than 1316. It is also clearly written here, under which circumstances these values hold.

We can remove this, but note that the true maximum payload size can only be determined after connection. As an alternative, we can have some tools to determine the payload size with foreseen IP version, AEAD and FEC.